test/unit/data-sources/rural-payments.test.js

```diff
@@ -8,59 +9,108 @@ const logger = {
 }
 
 describe('RuralPayments', () => {
-  test('handles RPP errors', () => {
-    // eslint-disable-next-line no-new
-    const rp = new RuralPayments({ logger })
-
-    const error = new Error('test error')
-    error.extensions = { response: { status: 400, headers: { get: () => 'text/html' } } }
-    const request = {}
-    const url = 'test url'
-
-    rp.didEncounterError(error, request, url)
-
-    expect(logger.error).toHaveBeenCalledWith('#datasource - Rural payments - request error', {
-      error,
-      request,
-      response: error.extensions.response,
-      code: RURALPAYMENTS_API_REQUEST_001
+  describe('fetch', () => {
+    const mockFetch = jest.spyOn(RESTDataSource.prototype, 'fetch')
+    const dummyRequest = { method: 'POST' }
+
+    beforeEach(() => {
+      mockFetch.mockClear().mockReset()
```

Comment:
reset does the same as clear https://jestjs.io/docs/mock-function-api#mockfnmockreset